src/sentry/grouping/component.py

```diff
@@ -70,14 +70,19 @@ def _walk_components(c, stack):
             return ' '.join(items[-1])
         return self.name or 'others'
 
-    def get_subcomponent(self, id):
+    def get_subcomponent(self, id, only_contributing=False):
         """Looks up a subcomponent by the id and returns the first or `None`."""
-        return next(self.iter_subcomponents(id), None)
+        return next(self.iter_subcomponents(
+            id=id,
+            only_contributing=only_contributing
+        ), None)
 
-    def iter_subcomponents(self, id, recursive=False):
+    def iter_subcomponents(self, id, recursive=False, only_contributing=False):
```

Comment:
This is a better solution because it takes into account if a stacktrace still contributes but an entire exception was set to not contribute.